### ISSUE TITLE
iOS App: French (Switzerland) and Italian (Switzerland) as additional languages for the spelling checker

### ISSUE THREAD

**Issue by Gianni154** [CollaboraOnline/online#1463]
**Describe the bug**
Is not really a bug. In Switzerland, French and Italian are among the national languages in addition to German. I noticed that only French (France) and Italian (Italy) are available for selection for spell checking.
Would it be possible to add French (Switzerland) and Italian (Switzerland) as additional language packs?

**Expected behavior**
The languages French (Switzerland) and Italian (Switzerland) are available for the spelling checker.

**Actual behavior**
Only French (France) and Italian (Italy) are available as choices for spell checking.

**Screenshots**

![File_000](https://user-images.githubusercontent.com/78140364/107618906-9a7e0400-6c52-11eb-89ec-a4b7ca1aaf50.jpeg)

**Device**
 - Device: iPad 6th generation 
 - OS: iPad OS 14.4
 - Version: iOS Collabora App Version 6.4.5 (2)

**Comment by wget** [comment]
Have checked into LibreOffice and it seems indeed we offer several versions of French. While the differences aren't that much between French variants, I know some French dialects like in Switzerland could have different formats for numbers and currencies. Considering this, I can understand your pain :)

![Screenshot_20210211_150711](https://user-images.githubusercontent.com/2997662/107647096-0d9b7080-6c7b-11eb-9367-b695f43ef34a.png)

**Comment by timar** [comment]
Currently we use spellckecker + dictionaries from iOS, so we have the language support that exists in iOS.

**Comment by tml1024** [comment]
Is the Italian (Italy) spell checked good enough as such for Italian (Switzerland)? Are there differences in orthography or vocabulary, are there Italian words used in Italy that are considered non-words in Switzerland?

And the same for French.

**Comment by tml1024** [comment]
As far as I can see, upstream LibreOffice uses just one dictionary for Italian. Unlike for German, there are no separate it_CH files:

```[tml@snorken core]$ ls -l dictionaries/it_IT/
total 3984
-rw-rw-r--  1 tml tml    1052 Oct 28 14:07 CHANGELOG.txt
-rw-rw-r--  1 tml tml     924 Oct 28 14:07 description.xml
-rw-rw-r--. 1 tml tml    1705 Jun 18  2018 dictionaries.xcu
-rw-rw-r--  1 tml tml    2154 Oct 28 14:07 hyph_it_IT.dic
-rw-rw-r--  1 tml tml   70054 Oct 28 14:07 it_IT.aff
-rw-r--r--  1 tml tml 1295078 Jan 21 11:52 it_IT.dic
drwxrwxr-x. 2 tml tml    4096 Jun 18  2018 legacy
drwxrwxr-x. 2 tml tml    4096 Jun 18  2018 META-INF
-rw-rw-r--  1 tml tml     949 Oct 28 14:07 README_hyph_it_IT.txt
-rw-rw-r--  1 tml tml   38897 Oct 28 14:07 README_it_IT.txt
-rw-rw-r--  1 tml tml   43382 Oct 28 14:07 README_th_it_IT.txt
-rw-rw-r--  1 tml tml 2126721 Oct 28 14:07 th_it_IT_v2.dat
-rw-rw-r--  1 tml tml  461580 Oct 28 14:07 th_it_IT_v2.idx
```

Ditto for French:

```[tml@snorken core]$ ls -l dictionaries/fr_FR/
total 6048
-rw-r--r--  1 tml tml     806 Dec 27 23:11 description.xml
-rw-rw-r--. 1 tml tml    1802 Jun 18  2018 dictionaries.xcu
-rw-r--r--  1 tml tml  201591 Dec 27 23:11 fr.aff
-rw-r--r--  1 tml tml 1227095 Dec 27 23:11 fr.dic
-rw-rw-r--. 1 tml tml   23253 Jun 18  2018 hyph_fr.dic
-rw-rw-r--. 1 tml tml   32328 Jun 18  2018 hyph-fr.tex
-rw-rw-r--. 1 tml tml     936 Jun 18  2018 icon.png
drwxrwxr-x. 2 tml tml    4096 Jun 18  2018 META-INF
-rw-r--r--  1 tml tml     434 Dec 27 23:11 package-description.txt
-rw-r--r--  1 tml tml    2758 Dec 27 23:11 README_fr.txt
-rw-rw-r--. 1 tml tml    2986 Jun 18  2018 README_hyph_fr.txt
-rw-rw-r--. 1 tml tml    2036 Jun 18  2018 README_thes_fr.txt
-rw-rw-r--. 1 tml tml 4668643 Jun 18  2018 thes_fr.dat
```

So I will fix this by simply adding code that claims the system it_IT dictionary is good for it_CH, too. For French I will claim the system fr_FR dictionary is good for not only Switzerland but the other countries where French is spoken, too. But that is probably more risky. Isn't the French in Quebec at least significantly different from French in France?

**Comment by tml1024** [comment]
Fix in https://gerrit.libreoffice.org/c/core/+/112670

**Comment by nikslor** [comment]
I mentioned in the Telegram/IRC chat on March 24), that the "German (Switzerland)" dictionary seems not yet to work. This seems fixed in 6.4.8 (2).

<img width="1112" alt="swiss_german_spellcheck" src="https://user-images.githubusercontent.com/1211300/112599247-fae69080-8e0f-11eb-9d0e-edbc16cff0d0.png">
